shopgate.php

```diff
@@ -74,845 +147,624 @@ public function __construct()
 
 		$this->displayName = $this->l('Shopgate');
 		$this->description = $this->l('Sell your products with your individual app and a website optimized for mobile devices.');
-
-		//delivery service list
-		$this->shipping_service_list = array
-		(
-			'OTHER' => $this->l('Other'),
-			'DHL' => $this->l('DHL'),
-			'DHLEXPRESS' => $this->l('DHL Express'),
-			'DP' => $this->l('Deutsche Post'),
-			'DPD' => $this->l('DPD'),
-			'FEDEX' => $this->l('FedEx'),
-			'GLS' => $this->l('GLS'),
-			'HLG' => $this->l('Hermes'),
-			'TNT' => $this->l('TNT'),
-			'TOF' => $this->l('trans-o-flex'),
-			'UPS' => $this->l('UPS'),
-			'LAPOSTE' => $this->l('LA POSTE'),
-		);
-
-		$this->product_export_descriptions = array
-		(
-			self::PRODUCT_EXPORT_DESCRIPTION => $this->l('Description'),
-			self::PRODUCT_EXPORT_SHORT_DESCRIPTION => $this->l('Short Description'),
-			self::PRODUCT_EXPORT_BOTH_DESCRIPTIONS => $this->l('Short Description + Description'),
-		);
-
-		$this->shopgate_trans = array(
-			'Bankwire' => $this->l('Bankwire'),
-			'Cash on Delivery' => $this->l('Cash on Delivery'),
-			'PayPal' => $this->l('PayPal'),
-			'Mobile Payment' => $this->l('Mobile Payment'),
-			'Shopgate' => $this->l('Shopgate')
-		);
-	}
-
-	private function setCarrier(Carrier $carrier)
-	{
-		$carrier->name = 'Shopgate';
-		$carrier->is_module = 1;
-		$carrier->deleted = 1;
-		$carrier->shipping_external = 1;
-
-		// 			if(version_compare(_PS_VERSION_, '1.4.4.0', '<') && version_compare(_PS_VERSION_, '1.4.2.5', '>=')){
-		// 				// fix a bug in Prestashop before version 1.4.4.0 classes/cart.php function isCarrierInRange() range behavior
-		// 				$carrier->shipping_method = Carrier::SHIPPING_METHOD_FREE;
-		// 			}
-		if (version_compare(_PS_VERSION_, '1.4.0.1', '<='))
-		{
-			// calculating shipping costs in Prestashop before version 1.4.4.2
-			//$carrier->shipping_method = Carrier::SHIPPING_METHOD_PRICE;
-			$carrier->range_behavior = 1;
-			$carrier->active = 1;
-			$carrier->shipping_handling = 0;
-		}
-
-		$carrier->external_module_name = 'shopgate';
-		foreach (Language::getLanguages() as $language)
-			$carrier->delay[$language['id_lang']] = $this->l('Depends on Shopgate selected carrier');
-	}
-
-
-	private function log($message, $type)
-	{
-		ShopgateLogger::getInstance()->log($message, $type);
+		$this->ps_versions_compliancy = array('min' => '1.4', 'max' => '1.6.99.99');
```

Comment:
The attribute ps_versions_compliancy can be safely removed. Your version range is quite wide and many PrestaShop 1.5 does not handle the values properly.